src/corvus_python/pyspark/utilities/dataframe_utils/null_safe_join.py

```diff
@@ -1,8 +1,12 @@
 """Copyright (c) Endjin Limited. All rights reserved."""
 
 from typing import List
-from pyspark.sql import DataFrame
-from pyspark.sql.functions import broadcast, expr
+try:
+    from pyspark.sql import DataFrame
+    from pyspark.sql.functions import broadcast, expr
+except ImportError as exc:
+    raise ImportError("PySpark is required for null_safe_join functionality. "
```

Comment:
Does this really have to be done in every file? Isn't there a catch-all way of doing it once in the top level `corvus_python.pyspark` init?

Reply:
Maybe. Let me try.